### PR TITLE
Implement sys_time_get_timezone syscall

### PR DESCRIPTION
**Pull request by Nekotekina**
Try to get system timezone automatically.
Used to be hardcoded to GMT+3.
Attempt to address #10195.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_time.cpp [hunk 1]
```diff
@@ -181,9 +183,15 @@ error_code sys_time_get_timezone(vm::ptr<s32> timezone, vm::ptr<s32> summertime)
...
-	*summertime = 0;
+#ifdef _WIN32
+	TIME_ZONE_INFORMATION tz{};
+	const DWORD tztype = GetTimeZoneInformation(&tz);
+	*timezone = tz.Bias;
+#else
+	*timezone = ::narrow<s32>(::timezone / 60);
+#endif
 
+	*summertime = 0;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_time.cpp:194]: you can probably use tztype to get the summertime info:
image

Comment (Nekotekina) [reply rpcs3/Emu/Cell/lv2/sys_time.cpp:194]: I'm not sure how its value is interpreted. Easier to keep it zero.

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_time.cpp:194]: then please add a TODO comment

**Comment by MarioSonic2987** [comment]
Fixes incorrect hour in God of War savedata (UTC +02:00 CEST):

**Before**:
image

**After**:
image

**Comment by Nekotekina** [comment]
Fixed summertime (looks like it's supplied in minutes), please retest.

File: rpcs3/Emu/Cell/lv2/sys_time.cpp [hunk 2]
```diff
@@ -179,10 +183,46 @@ u64 get_guest_system_time()
...
+	}
+	default:
+	{
+		ensure(0);
+	}
+	}
+#elif __linux__
+	tzset();
+	*timezone = ::narrow<s32>(-::timezone / 60);
+	*summertime = ::daylight ? 60 : 0;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_time.cpp:218]: I wonder if this should be negative.

File: rpcs3/Emu/Cell/lv2/sys_time.cpp [hunk 3]
```diff
@@ -179,10 +183,46 @@ u64 get_guest_system_time()
...
+	case TIME_ZONE_ID_UNKNOWN:
+	{
+		*timezone = -tz.Bias;
+		*summertime = 0;
+		break;
+	}
+	case TIME_ZONE_ID_STANDARD:
+	{
+		*timezone = -tz.Bias;
+		*summertime = -tz.StandardBias;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_time.cpp:201]: I don't really understand the standard bias. But I'd assume that it should be added to the bias and not counted as summertime.

File: rpcs3/Emu/Cell/lv2/sys_time.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_time.cpp:218]: ah I missed the - in front of the bias. nevermind

**Pull request merged**